### PR TITLE
refactor(core)!: drop `Mode` directive

### PR DESCRIPTION
**Pull request by waterplea**
Closes #

### CODE REVIEW

**Review by splincode — APPROVED** [review]

File: projects/core/components/tooltip/tooltip.component.ts [hunk 1]
```diff
@@ -9,22 +9,17 @@ import {
...
-import type {TuiBrightness} from '@taiga-ui/core/types';
-import {takeUntil} from 'rxjs';
 
 @Component({
     selector: 'tui-tooltip',
     templateUrl: './tooltip.template.html',
     styleUrls: ['./tooltip.style.less'],
     changeDetection: ChangeDetectionStrategy.OnPush,
-    providers: [TuiDestroyService, MODE_PROVIDER],
+    providers: [TuiDestroyService],
```
Comment (vladimirpotekhin) [inline projects/core/components/tooltip/tooltip.component.ts:18]: nit: Is `TuiDestroyService` now redundant?

File: projects/demo/src/modules/directives/theme/examples/1/index.html [hunk 2]
```diff
@@ -16,8 +16,8 @@
         <tui-input-number [(ngModel)]="money">Sum</tui-input-number>
     </p>
     <div
+        tuiTheme="light"
         class="light"
```
Comment (vladimirpotekhin) [inline projects/demo/src/modules/directives/theme/examples/1/index.html:20]: there is a white text on white background in this example
seems to be due to inherit color in input
Снимок экрана 2024-04-09 в 10 53 21
Снимок экрана 2024-04-09 в 10 54 14

**Review by vladimirpotekhin — APPROVED** [review]

File: projects/demo/src/modules/directives/theme/examples/1/index.html (hunk 2 above)
Comment (vladimirpotekhin) [reply projects/demo/src/modules/directives/theme/examples/1/index.html:20]: and too dark hints 

Снимок экрана 2024-04-09 в 11 13 57

File: projects/core/components/tooltip/tooltip.component.ts (hunk 1 above)
Comment (waterplea) [reply projects/core/components/tooltip/tooltip.component.ts:18]: Looks like it. I'll remove it later.

**Pull request merged**